mapillary.py

```diff
@@ -94,45 +144,88 @@ def pull_and_save_trace_for_bbox(bbox: str) -> None:
+                continue
 
-    print('Getting seq for bbox={}'.format(bbox))
-    next_url = SEQUENCE_URL.format(map_client_id, bbox, seq_per_page)
-    page = 1
-    while next_url:
-        print('Page {}, url={}'.format(page, next_url))
-        sequence_resp = session_.get(next_url)
-        sequence_keys = []
-        for seq_f in sequence_resp.json()['features']:
             # Skip sequences that have too few images
-            if len(seq_f['geometry']['coordinates']) >= skip_if_fewer_imgs_than:
-                sequence_keys.append(seq_f['properties']['key'])
-        # TODO: Paginate images
-        images_resp = session_.get(IMAGES_URL.format(map_client_id, ','.join(sequence_keys)))
-        for img_f in images_resp.json()['features']:
-            if img_f['properties']['sequence_key'] not in result:
-                result[img_f['properties']['sequence_key']] = []
-            result[img_f['properties']['sequence_key']].append((
-                parser.isoparse(img_f['properties']['captured_at']).timestamp(),  # Epoch time
-                img_f['geometry']['coordinates']
-            ))
-
-        # Check if there is a next page or if we are finished with this bbox
-        next_url = sequence_resp.links['next']['url'] if 'next' in sequence_resp.links else None
-        page += 1
-
-    stop = time.time()
-    elapsed = stop - start
-    print('\n\n##################\nFinished processing seqs for bbox={}, elapsed time: {}'.format(bbox, elapsed))
-    print('{}\n##################\n\n'.format(result))
-
-    return result
+            if len(seq_f['geometry']['coordinates']) < skip_if_fewer_imgs_than:
+                continue
+
+            seq_ids.append(seq_id)
+
+        if len(seq_ids) > 0:
+            # Paginate images within these sequences
+            img_next_url = IMAGES_URL.format(map_client_id, ','.join(seq_ids), img_per_page)
+            img_page = 1
+            while img_next_url:
+                print('@@@ MAPILLARY: Image Page {}, url={}'.format(img_page, img_next_url))
+                img_resp = session_.get(img_next_url, timeout=10)
+                for img_f in img_resp.json()['features']:
+                    if img_f['properties']['sequence_key'] not in sequences_by_id:
+                        sequences_by_id[img_f['properties']['sequence_key']] = []
+                    sequences_by_id[img_f['properties']['sequence_key']].append({
+                        'time': parser.isoparse(img_f['properties']['captured_at']).timestamp(),  # Epoch time
+                        'lon': img_f['geometry']['coordinates'][0],
+                        'lat': img_f['geometry']['coordinates'][1]
+                    })
+
+                # Check if there is a next image page or if we are finished with this sequence
+                img_next_url = img_resp.links['next']['url'] if 'next' in img_resp.links else None
+                img_page += 1
```

Comment:
Adds pagination to `/images` call. Also transforms each trace datapoint to the format that Valhalla map matching (and the new filtering function) expects (lines 207-211).